eng/common/pipelines/templates/steps/verify-resource-ref.yml

```diff
@@ -0,0 +1,32 @@
+
+steps:
+  - pwsh: |
+      Install-Module -Name powershell-yaml -RequiredVersion 0.4.1 -Force -Scope CurrentUser
+      $remoteHEAD = git rev-parse origin/master
+      $localHEAD = git rev-parse HEAD
+      $ymlfilesChanged = (git diff --name-only $remoteHEAD...$localHEAD) | Where-Object {$_ -like '*.yml'}
```

Comment:
I would avoid the git diffing here as I think it adds a layer of complexity that isn't needed. I would simply take a directory path and find all yml files under it and check them.